src/ecs/commands.ts

```diff
@@ -0,0 +1,182 @@
+}
+
+export enum EcsRunCommandPrompt {
+    Enable = 'ecsRunCommandEnable',
+    Disable = 'ecsRunCommandDisable',
+}
+
+export async function toggleExecuteCommandFlag(
+    service: Service,
+    window = vscode.window,
+    settings = PromptSettings.instance
+): Promise<void> {
+    const yes = localize('AWS.generic.response.yes', 'Yes')
+    const yesDontAskAgain = localize('AWS.message.prompt.yesDontAskAgain', "Yes, don't ask again")
+    const no = localize('AWS.generic.response.no', 'No')
+
+    const isNotEnabled = service.description.enableExecuteCommand ?? false
+    const prompt = isNotEnabled ? EcsRunCommandPrompt.Enable : EcsRunCommandPrompt.Disable
+
+    const warningMessage = isNotEnabled
+        ? localize(
+              'AWS.command.ecs.runCommandInContainer.warning.enableExecuteFlag',
+              'Enabling command execution will change the state of resources in your AWS account, including but not limited to stopping and restarting the service.\nAltering the state of resources while the Execute Command is enabled can lead to unpredictable results.\n Continue?'
+          )
+        : localize(
+              'AWS.command.ecs.runCommandInContainer.warning.disableExecuteFlag',
+              'Disabling command execution will change the state of resources in your AWS account, including but not limited to stopping and restarting the service.\n Continue?'
+          )
+
+    if (await settings.isPromptEnabled(prompt)) {
+        const choice = await window.showWarningMessage(warningMessage, yes, yesDontAskAgain, no)
+        if (choice === undefined || choice === no) {
+            throw new CancellationError('user')
+        } else if (choice === yesDontAskAgain) {
+            settings.disablePrompt(prompt)
+        }
+    }
+
+    await service.toggleExecuteCommand()
+}
+
+export const runCommandInContainer = Commands.register('aws.ecs.runCommandInContainer', (obj?: unknown) => {
+    return telemetry.ecs_runExecuteCommand.run(async span => {
+        span.record({ ecsExecuteCommandType: 'command' })
+
+        const { container, task, command } = await runCommandWizard(obj)
+        const timeout = new Timeout(600000)
+        showMessageWithCancel('Running command...', timeout)
+
+        try {
+            const { path, args, dispose } = await container.prepareCommandForTask(command, task)
+            showOutputMessage(
+                `${moment().format(INSIGHTS_TIMESTAMP_FORMAT)}:  Container: "${
+                    container.description.name
+                }" Task ID: "${task}" Command: "${command}"`,
+                globals.outputChannel
+            )
+
+            const proc = new ChildProcess(path, args, { logging: 'noparams' })
+            await proc
+                .run({
+                    timeout,
+                    rejectOnError: true,
+                    rejectOnErrorCode: true,
+                    // TODO: `showOutputMessage` should not be writing to the logs...
```

Comment:
I assume the intention there was to get visibility in CI. But it's probably better to explicitly call getLogger() instead of trying to be clever.